### DISCUSSION TITLE
T18 HD touchscreen support

### DISCUSSION THREAD

**Discussion by Ga1j1n** [EdgeTX/edgetx#341] [General]
I asked a few questions in the AliExpress jumperstore and it seems a touchscreen variant of the T18 is due in the next 2 weeks, hopefully including the promised 800x480 screen promised for 2.4.

Assuming this comes to be and it is possible to buy an upgrade screen for existing T18's will it be supported in edgetx as is? I'd expect higher res displays to be the next logical evolution once stability is achieved, this could provide a a platform to develop on.

**Comment by raphaelcoeffic** [comment]
@Ga1j1n not sure what Jumper or whoever is selling this is doing, but there is not current plan for supporting this. This would require quite some changes, as all the bitmaps would be just to small to be seen/touched properly, and I'm not sure the RAM will be enough for everything then. I had already received such an HD display from Jumper, but really, the chance we will support it soon is very thin, if even existing.

**Reply by Ga1j1n** [reply]
That's fair, it's entirely possible the screen used in the end is the same 480x272 screen used in the Radioking/ eachine line as that shares main board etc with the T18.

Next question then, If I fitted a spare part radiomaster touchscreen will.it work?

**Reply by pfeerick** [reply]
Raphael will be able to say for certain, but I believe the T18 build is touchscreen ready, i.e yes.

**Reply by raphaelcoeffic** [reply]
The main issue here is the following: The screen mounted in the T18 is not the same as the TX16S. It has different parameters with respect to they way it is driven. So this means that a TX16S LCD would work, but you would need to "make your own target", using the T18 settings for everything but the screen.

**Reply by pfeerick** [reply]
What about the digitizer panel to make it a touch screen? Can you not add that on the T18 to make it a touch screen like you can with the TX16S (which is why mine is a touch-screen as it didn't come like that).

**Reply by raphaelcoeffic** [reply]
This is the same. If you can find it, that would be the path with the fewest friction. Jumper confirmed in btw. they won‘t offer an upgrade, but a new radio with touch.

**Reply by Ga1j1n** [reply]
Well that's a bit of a blow as I'd ordered a spare part TX16s touchscreen panel on spec, to swap out. If this can be resolved in firmware I'm more than happy to test drive if a dev is willing to build it?
Hopefully the effort also would benefit all the Radioking Tx18S and eachine TX16s users too as they are essentially the same radio / mainboards / layout.

Thanks for looking into it

**Reply by Ga1j1n** [reply]
> What about the digitizer panel to make it a touch screen? Can you not add that on the T18 to make it a touch screen like you can with the TX16S (which is why mine is a touch-screen as it didn't come like that).

I don't suppose you could identify the appropriate digitizer, link to it?

**Reply by pfeerick** [reply]
[This is what I used](https://www.banggood.com/Original-RadioMaster-Touch-Panel-Update-Set-Replacement-Parts-for-TX16S-Transmitter-p-1742124.html) for the TX16S - which at first glance would looks 'similar' to what I'd expect the T18 to use - given it also uses a 6 pin ribbon connector I2C touch screen controller - but I don't know if the similarity ends there - as they have completely redesigned the board layout on the T18, and I don't know if the ribbon on that touch panel would even reach the connector on the T18, let alone be the same pinout, or if the shape of the bezel is just that bit different. It's also very annoying Jumper referred  to it as the "800 * 480 HD IPS touch screen interface reserved" when they ship with a 480×272 screen and then it seems can't be bothered following through with that promise of an upgrade option.

**Reply by Ga1j1n** [reply]
So, as my brand new T18 arrived and so did the Tx16s spare screen I felt a test was in order after flashing edgeTx latest.
I can report that the screen does indeed work perfectly when connected alone but adding the digitzer to the mix means the radio will not boot.
I'm unsure if this can be fixed in the firmware or a different digitiser is required, connection wise everything matches up and I see no alignment issues although I tested the screen loose.
I wish I could buy an Eachine or Radioking spare to test as they are clearly OEM variants of the Jumper hardware and already feature touch screens.
Not sure where to go next, any thoughts?

**Reply by Ga1j1n** [reply]
Looks like the T18 Pro V2 is in the wild, here's hoping you will be supporting it!

https://www.aliexpress.com/item/1005002879767056.html?spm=a2g0o.productlist.0.0.67df8eb4T8uQou&algo_pvid=414b5783-7934-4099-bb93-d5d06a6d1042&algo_exp_id=414b5783-7934-4099-bb93-d5d06a6d1042-3

**Reply by raphaelcoeffic** [reply]
Should be ok, this should be the same as mine.

**Reply by pfeerick** [reply]
Oh, interesting... removal of the 5IN1 and reversion to the 4IN1 is a 'feature' (which I actually feel *is* an improvement, btw)... 😆

**Reply by raphaelcoeffic** [reply]
I think the main issue with MPM 4IN-whatever is “availability”. As you, most of chips on that module are actually end-of-life…
